### PR TITLE
Drop numpy as a dev dependency

### PR DESCRIPTION
**Pull request by trickeydan**
## Checklist

- [x] I have updated the relevant documentation
- [x] I have added the `semver-major`, `semver-minor` or `semver-patch` label as appropriate
- [ ] I would like multiple reviewers to approve my code before merging

## Why do you want to make these changes?

This is now a dependency in j5-zoloto, so it is not needed here.

Also, we were locking it to a sufficiently old version of numpy that it wouldn't install the dependencies on darwin-arm64.

## Which parts of the codebase do your changes affect?

None

## How do your changes affect student or volunteer experience?

M1 Macs can now be used for development

## Are your changes related to any existing issues or PRs? How so?

Nope.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = "^3.7"
```
Comment (RealOrangeOne) [inline pyproject.toml:20]: Question: Is this change intentional?

Comment (trickeydan) [reply pyproject.toml]: Fixed and force-pushed :P

**Review by RealOrangeOne — APPROVED** [review]

**Review by raccube — APPROVED** [review]
